src/passes/pass.cpp

```diff
@@ -192,7 +193,7 @@ void PassRunner::run() {
       std::cerr << "final module does not validate\n";
       abort();
     }
-    passDebug = currPassDebug;
+    recursion--;
```

Comment:
A problem though is that it means code can't easily be moved around, as we might need to add `nested` to the name. A concrete example is printing: we might call print as a toplevel thing, in wasm-dis for example (when it prints out the output), but we also call print in other cases, some of which happen to be inside other passes.

So I don't think this is optimal either way, as I agree with your criticism of the current code in this PR. However, as this is purely meant for `BINARYEN_PASS_DEBUG`, a weird debug mode, I am not too worried about cleanliness in this code itself. While adding a nested version means lack of cleanliness outside, which worries me more.

However, perhaps the nested version makes sense for other reasons?